Compose hover request that is outside of the file will cause an error

https://github.com/docker/docker-language-server/blob/e86df36de1dbbc3b62f5739f8b6598305bea9b5f/internal/compose/hover.go#L50

```
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1

goroutine 312 [running]:
testing.tRunner.func1.2({0x103e11d80, 0x140000443c0})
	/usr/local/go/src/testing/testing.go:1734 +0x1ac
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1737 +0x334
panic({0x103e11d80?, 0x140000443c0?})
	/usr/local/go/src/runtime/panic.go:792 +0x124
github.com/docker/docker-language-server/internal/compose.Hover({0x140002aa930?, 0x1400015c320?}, 0x140005abf38, {0x103ebf0c8, 0x140002a6580})
	.../docker-language-server/internal/compose/hover.go:50 +0x308
github.com/docker/docker-language-server/internal/compose.TestHover_SchemaDocumentation.func1(0x14000215180)
	.../docker-language-server/internal/compose/hover_test.go:394 +0xc4
testing.tRunner(0x14000215180, 0x14000295360)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ All notable changes to the Docker Language Server will be documented in this fil
     - consider quotes when calculating the link's range ([#242](https://github.com/docker/docker-language-server/issues/242))
   - textDocument/hover
     - prevent YAML hover issues caused by whitespace ([#244](https://github.com/docker/docker-language-server/issues/244))
+    - ignore hover requests that are outside the file to prevent panics ([#261](https://github.com/docker/docker-language-server/issues/261))
   - textDocument/prepareRename
     - fix range calculation when the element is quoted ([#255](https://github.com/docker/docker-language-server/issues/255))
   - textDocument/rename

--- a/e2e-tests/hover_test.go
+++ b/e2e-tests/hover_test.go
@@ -156,6 +156,12 @@ func testHover_Compose(t *testing.T, composeSupport bool) {
 				},
 			},
 		},
+		{
+			name:     "hovering outside the file",
+			content:  "version: 1.2.3",
+			position: protocol.Position{Line: 1, Character: 4},
+			result:   nil,
+		},
 	}
 
 	for _, tc := range testCases {
@@ -168,7 +174,7 @@ func testHover_Compose(t *testing.T, composeSupport bool) {
 			err = conn.Call(context.Background(), protocol.MethodTextDocumentHover, protocol.HoverParams{
 				TextDocumentPositionParams: protocol.TextDocumentPositionParams{
 					TextDocument: protocol.TextDocumentIdentifier{URI: didOpen.TextDocument.URI},
-					Position:     protocol.Position{Line: 0, Character: 3},
+					Position:     tc.position,
 				},
 			}, &hover)
 			require.NoError(t, err)

--- a/internal/compose/hover.go
+++ b/internal/compose/hover.go
@@ -23,6 +23,9 @@ func Hover(ctx context.Context, params *protocol.HoverParams, doc document.Compo
 	line := int(params.Position.Line) + 1
 	character := int(params.Position.Character) + 1
 	lines := strings.Split(string(doc.Input()), "\n")
+	if line > len(lines) {
+		return nil, nil
+	}
 
 	for _, documentNode := range file.Docs {
 		if mappingNode, ok := documentNode.Body.(*ast.MappingNode); ok {

--- a/internal/compose/hover_test.go
+++ b/internal/compose/hover_test.go
@@ -378,6 +378,13 @@ services:
 				},
 			},
 		},
+		{
+			name:      "hovering outside the file",
+			content:   "version: 1.2.3",
+			line:      1,
+			character: 4,
+			result:    nil,
+		},
 	}
 
 	composeFile := fmt.Sprintf("file:///%v", strings.TrimPrefix(filepath.ToSlash(filepath.Join(os.TempDir(), "compose.yaml")), "/"))

